drivers/include/periph/adc.h

```diff
@@ -134,20 +134,57 @@ int32_t adc_sample(adc_t line, adc_res_t res);
  * @note requires the `periph_adc_continuous` feature
  *
  * @param[in] res           resolution to use for conversion
+ * @param[in] f_adc         frequency to use for conversion, leave 0 for default
  */
-void adc_continuous_begin(adc_res_t res);
+void adc_continuous_begin(adc_res_t res, uint32_t f_adc);
 
 /**
- * @brief   Sample an ADC line without powering off the ADC afterward
+ * @brief   Sample an ADC line without powering off the ADC afterwards
  *
  * @note requires the `periph_adc_continuous` feature
  *
- * @brief   Sample a value from the given ADC line
+ * @param[in] line          line to sample
  *
  * @return                  the sampled value on success
  */
 int32_t adc_continuous_sample(adc_t line);
 
+/**
+ * @brief   Capture multiple ADC samples from an ADC line
+ *
+ * @note requires the `periph_adc_continuous` feature
+ *
+ * @param[in]  line         line to sample
+ * @param[out] buf          target buffer
+ * @param[in]  len          number of samples to sample
+ */
+void adc_continuous_sample_multi(adc_t line, uint16_t *buf, size_t len);
+
+/**
+ * @brief   Configure the ADC with a given resolution for continuous sampling
+ *          on two lines.
+ *
+ * @note requires the `periph_adc_continuous` feature
+ *
+ * @param[in] res           resolution to use for conversion
+ * @param[in] f_adc         frequency to use for conversion, leave 0 for default
+ */
+void adc_dual_continuous_begin(adc_res_t res, uint32_t f_adc);
+
+/**
+ * @brief   Capture multiple ADC samples from two ADC lines at the same time
+ *
+ * @note The hardware needs to be capable of sampling two lines at the same time,
+ *       e.g. they need to be configured to two separate ADC instances.
+ *
+ * @note requires the `periph_adc_continuous` feature
+ *
+ * @param[in]  line         lines to sample
+ * @param[out] buf          target buffers
+ * @param[in]  len          number of samples to sample
+ */
+void adc_dual_continuous_sample_multi(adc_t line[2], uint16_t *buf[2], size_t len);
```

Comment:
if multi is just 2 it should have a name that is indicating that 
```suggestion
void adc_dual_continuous_sample_2(adc_t line[2], uint16_t *buf[2], size_t len);
```

```suggestion
void adc_dual_continuous_sample_double(adc_t line[2], uint16_t *buf[2], size_t len);
```

```suggestion
void adc_dual_continuous_sample_stereo(adc_t line[2], uint16_t *buf[2], size_t len);
```